app/views/pulitzer/versions/_edit.html.erb

```diff
@@ -0,0 +1,41 @@
+<% if version.processing_failed? %>
+  <div class="pulitzer-row">
+    There was a problem creating the new preview version of this post for editing.
+    <ul>
+      <% version.cloning_errors.each do |error| %>
+        <li><%= error %></li>
+      <% end %>
+    </ul>
+  </div>
+<% elsif version.processing? %>
+  <h2> Processing Preview for <%= version.title %> </h2>
+  <div class="pulitzer-row">
+    <progress value="0" max="<%= version.post.active_version.content_elements.count + version.post.active_version.post_tags.count + 2 %>" <%= listener_attrs version, :processed_element_count if defined? ForeignOffice %>></progress>
```

Comment:
It's actually a bit silly, I just wanted to count as many things as possible to keep the progress meter moving smoothly. So I count each content element as it's copied, then I count each post tag as it's copied, then I count once for updating the status in the database, and once more for generating the url for the new preview version.